src/main/java/org/opensearch/agent/tools/SearchAnomalyDetectorsTool.java

```diff
@@ -123,24 +131,92 @@ public <T> void run(Map<String, String> parameters, ActionListener<T> listener)
+                        });
+
+                    GetAnomalyDetectorRequest profileRequest = new GetAnomalyDetectorRequest(
+                        hit.getId(),
+                        Versions.MATCH_ANY,
+                        true,
+                        true,
+                        "",
+                        "",
+                        false,
+                        null
+                    );
+                    adClient.getDetectorProfile(profileRequest, profileListener);
+                }
+                CompletableFuture<List<GetAnomalyDetectorResponse>> listFuture = CompletableFuture
+                    .allOf(profileFutures.toArray(new CompletableFuture<?>[0]))
+                    .thenApply(v -> profileFutures.stream().map(CompletableFuture::join).collect(Collectors.toList()));
+                final List<GetAnomalyDetectorResponse> profileResponses = listFuture.join();
+                for (GetAnomalyDetectorResponse profileResponse : profileResponses) {
+                    if (profileResponse != null && profileResponse.getDetector() != null) {
+                        String detectorId = profileResponse.getDetector().getId();
+
+                        // We follow the existing logic as the frontend to determine overall detector state
+                        // https://github.com/opensearch-project/anomaly-detection-dashboards-plugin/blob/main/server/routes/utils/adHelpers.ts#L437
+                        String detectorState;
+                        ADTask realtimeTask = profileResponse.getRealtimeAdTask();
+
+                        if (realtimeTask != null) {
+                            String taskState = realtimeTask.getState();
+                            if (taskState.equalsIgnoreCase("CREATED")) {
+                                detectorState = DetectorStateString.Initializing.name();
+                            } else if (taskState.equalsIgnoreCase("RUNNING")) {
+                                detectorState = DetectorStateString.Running.name();
+                            } else if (taskState.equalsIgnoreCase("INIT_FAILURE")
+                                || taskState.equalsIgnoreCase("UNEXPECTED_FAILURE")
+                                || taskState.equalsIgnoreCase("FAILED")) {
+                                detectorState = DetectorStateString.Failed.name();
+                            } else {
+                                // Task states may fall under other values, such as "FEATURE_REQUIRED" / "STOPPED" / etc.
+                                // Defaulting all other states to fall under the disabled category
+                                detectorState = DetectorStateString.Disabled.name();
+                            }
+                        } else {
+                            detectorState = DetectorStateString.Disabled.name();
+                        }
+
+                        if (running != null && !running && detectorState.equals(DetectorStateString.Running.name())) {
```

Comment:
(Style) Here and two below.  If you use `Boolean.FALSE.equals(running)` it handles the null check.

Reply:
Cool! thanks for the suggestion.

Reply:
Added